content/source/docs/enterprise/sentinel/import/index.html.md

```diff
@@ -24,100 +24,161 @@ see [Mocking Terraform Sentinel Data](../mock.html).
+## Useful Functions and Idioms for Terraform Sentinel Policies
 
-Terraform's internal data formats are complex, which means basic Sentinel policies for Terraform are more verbose than basic policies that use simpler data sources.
-
-This will improve in future versions of Terraform and Sentinel; in the meantime, be aware of the following idioms as you start writing policies for Terraform.
+The following functions and idioms will be useful as you start writing Sentinel policies for Terraform.
 
 ### To Find Resources, Iterate over Modules
 
-The most basic Sentinel task for Terraform is to enforce a rule on all resources of a given type. Before you can do that, you need to get a collection of all the relevant resources.
-
-The easiest way to do that is to copy a function like the following into any policies that examine every resource in a configuration:
+The most basic Sentinel task for Terraform is to enforce a rule on all resources of a given type. Before you can do that, you need to get a collection of all the relevant resources. The easiest way to do that is to copy a function like the following into your policies:
 
 ```python
 import "tfplan"
-
-# Get an array of all resources of the given type (or an empty array).
-get_resources = func(type) {
-	if length(tfplan.module_paths else []) > 0 { # always true in the real tfplan import
-		return get_resources_all_modules(type)
-	} else { # fallback for tests
-		return get_resources_root_only(type)
-	}
-}
-
-get_resources_root_only = func(type) {
-	resources = []
-	named_and_counted_resources = tfplan.resources[type] else {}
-	# Get resource bodies out of nested resource maps, from:
-	# {"name": {"0": {"applied": {...}, "diff": {...} }, "1": {...}}, "name": {...}}
-	# to:
-	# [{"applied": {...}, "diff": {...}}, {"applied": {...}, "diff": {...}}, ...]
-	for named_and_counted_resources as _, instances {
-		for instances as _, body {
-			append(resources, body)
-		}
-	}
-	return resources
-}
-
-get_resources_all_modules = func(type) {
-	resources = []
-	for tfplan.module_paths as path {
-		named_and_counted_resources = tfplan.module(path).resources[type] else {}
-		# Get resource bodies out of nested resource maps, from:
-		# {"name": {"0": {"applied": {...}, "diff": {...} }, "1": {...}}, "name": {...}}
-		# to:
-		# [{"applied": {...}, "diff": {...}}, {"applied": {...}, "diff": {...}}, ...]
-		for named_and_counted_resources as _, instances {
-			for instances as _, body {
-				append(resources, body)
-			}
-		}
-	}
-	return resources
+import "strings"
+
+# Find all resources of a specific type from all modules using the tfplan import
+find_resources_from_plan = func(type) {
+        resource_maps = {}
+        for tfplan.module_paths as path {
+                # Compute joined_path from the module path
+                if length(path) == 0 {
+                        joined_path = ""
+                } else {
+                        joined_path = "module." + strings.join(path, ".module.") + "."
```

Comment:
This line will not print a correct resource address as `path` may have multiple entries, producing something like `module.foo.bar.module.`

A helper would probably be better here after all:

```
module_address = func(path) {
	if length(path) < 1 {
		return ""
	}

	result = []
	for path as p {
		append(result, ["module", "p"]) // Can also be multiple append calls
	}

	return strings.join(result, ".")
}
```

Reply:
Yeah I see that now - sorry about that!